theia-ide/vscode-builtin-extensions
# Fix build, update dependencies

### What it does

#### Example applications dependencies refresh

yarn.lock was removed and regenerated to pick-up a more recent version of dependencies, including Theia extensions. 

I believe that this update broke Electron, because of upstream changes. This was fixed by adding `@theia/electron` as a dev-dependency in the Electron Example application. 

#### Build
ATM we need node option "-max-old-space-size=8192" to be set for the build to succeed, when one runs "yarn" on the repo. Else we get errors like this:
```
<--- Last few GCs --->

[13559:0x2ed8a10]    45844 ms: Mark-sweep 1384.4 (1424.4) -> 1383.3 (1423.9) MB, 574.8 / 0.0 ms  (average mu = 0.154, current mu = 0.111) allocation failure scavenge might not succeed
[13559:0x2ed8a10]    45862 ms: Scavenge 1384.3 (1423.9) -> 1383.3 (1423.9) MB, 3.8 / 0.0 ms  (average mu = 0.154, current mu = 0.111) allocation failure 
[13559:0x2ed8a10]    45881 ms: Scavenge 1384.3 (1423.9) -> 1383.4 (1424.4) MB, 3.9 / 0.0 ms  (average mu = 0.154, current mu = 0.111) allocation failure 


<--- JS stacktrace --->

==== JS stack trace =========================================

Security context: 0x18391c41e6e9 <JSObject>
    0: builtin exit frame: captureStackTrace(this=0x18391c409719 <JSFunction Error (sfi = 0x3d5829a16e01)>,0x04afcfcfde11 <JSFunction uvException (sfi = 0x18391c450501)>,0x23789eb07a29 <Error map = 0x1652bc3d48f9>,0x18391c409719 <JSFunction Error (sfi = 0x3d5829a16e01)>)

    1: uvException [0x4afcfcfde11] [internal/errors.js:285] [bytecode=0x36070f28efd1 offset=817](this=0x04afcfc84199 <Object ...


FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory

```

Using this node option was already done in root package.json, but it seems the option was not being passed-down to sub-scripts, and so the build still fails, for some built-in extensions.
    
I played with this a bit and found how that if I add the node option directly for scripts "compile:extensions" and "bundle:extensions", then it works.

### How to test
Clone a fresh copy of the repo and try building it locally using `yarn` on master. Observe that there are some failures when building VS Code built-in extensions.

Then try again using this PR, and confirm that the failures are gone.

Try the example applications to confirm that they still build, start and generally work.

Signed-off-by: Marc Dumais <marc.dumais@ericsson.com>
